crates/bevy_render/src/renderer/render_device.rs

```diff
@@ -101,19 +101,24 @@ impl RenderDevice {
 
     /// Creates a [`RenderPipeline`].
     #[inline]
-    pub fn create_render_pipeline(&self, desc: &RawRenderPipelineDescriptor) -> RenderPipeline {
+    pub(crate) fn create_render_pipeline(
+        &self,
+        id: RenderPipelineId,
+        desc: &RawRenderPipelineDescriptor,
+    ) -> RenderPipeline {
         let wgpu_render_pipeline = self.device.create_render_pipeline(desc);
-        RenderPipeline::from(wgpu_render_pipeline)
+        RenderPipeline::new(id, wgpu_render_pipeline)
     }
```

Comment:
This change might be slightly controversial. Pipelines can only be created by the pipeline cache.